bundles/admin/admin-layereditor/view/LocalizingFlyout.jsx

```diff
@@ -0,0 +1,175 @@
+import { LocaleContext, handleBinder, StateHandler, mutatorMixin, Mutator } from 'oskari-ui/util';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+
+const ExtraFlyout = Oskari.clazz.get('Oskari.userinterface.extension.ExtraFlyout');
+
+export class LocalizingFlyout extends ExtraFlyout {
+    constructor (instance, title, options = {}) {
+        super(title, options);
+        this.instance = instance;
+        this.addClass('admin-localizing-flyout');
+        this.mountPoint = document.createElement('div');
+        this.setContent(this.mountPoint);
+
+        // Create a handler for ui state.
+        this.uiHandler = new UIHandler(instance);
+        // Calling updateState will affect only the properties listed.
+        this.uiHandler.updateState({ headerMessageKey: options.headerMessageKey });
+
+        // Bind methods starting with 'on'.
+        handleBinder(this, 'on');
+        // Now they won't lose their context when called by something else than this LocalizingFlyout.
+        this.on('show', this.onShow);
+        this.on('hide', this.onHide);
+        this.uiHandler.setCancelAction(this.onCancel);
+        // State listener is called every time the state changes.
+        // Here we are telling the UI to re render on state change, making the UI respond to user actions.
+        this.uiHandler.addStateListener(this.onUpdate);
+    }
+    setAction (action) {
+        this.uiHandler.setSaveAction(action);
+    }
+    setLoading (loading) {
+        this.uiHandler.updateState({ loading: !!loading });
+    }
+    onCancel () {
+        this.hide();
+    }
+    onShow () {
+        // The UI handler may provide functions that can be accessed outside React Elements.
+        this.uiHandler.reset();
+    }
+    onHide () {
+        ReactDOM.unmountComponentAtNode(this.mountPoint);
+    }
+    onUpdate () {
+        // The UI handler itself should never be passed to a component.
+        // Instead, the handler should provide a mutator object, a subset of the handlers methods.
+        // The mutator should contain only methods the components require.
+        const mutator = this.handler.getMutator();
+        let ui = (
+            <LocaleContext.Provider value={{ bundleKey: this.instance.getName() }}>
+                <LocalizedContent { ...this.handler.getState() } mutator={mutator}/>
+            </LocaleContext.Provider>
+        );
+        ReactDOM.render(ui, this.mountPoint);
+    }
+}
+
+// Create a service responsible of the UI state.
+class UIService extends StateHandler {
+    constructor (instance) {
+        super();
+        this.getMsg = Oskari.getMsg.bind(null, instance.getName());
+        this.setState({
+            loading: false,
+            headerMessageKey: null,
+            value: {},
+            labels: {
+                en: this.getMsg('en.lang'),
+                fi: this.getMsg('fi.lang'),
+                sv: this.getMsg('sv.lang')
```

Comment:
Hardcoded languages. These should come from the Oskari env.

Reply:
Using Oskari.getSupportedLanguages